File: result/metagpt/claude/humaneval/35.c
#include <stdio.h>
#include <limits.h> // 用于定义浮点数的最小值

float max_element(float* l, int size) {
    float max = -FLT_MAX; // 使用浮点数的最小值作为初始值
    for (int i = 0; i < size; i++) {
        if (max < l[i]) {
            max = l[i];
        }
    }
    return max;
}

File: result/metagpt/deepseek/humaneval/35.rs
pub fn max_element(l: &[f32]) -> f32 {
    l.iter().copied().fold(f32::NEG_INFINITY, f32::max)
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_max_element() {
        assert_eq!(max_element(&[1.0, 2.0, 3.0]), 3.0);
        assert_eq!(max_element(&[-1.0, -2.0, -3.0]), -1.0);
    }
}